orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_data_files.py

```diff
@@ -1,12 +1,236 @@
+            mapping_key=blob.name.replace('/', '_').replace('-', '_')  # filter out bad chars for dagster dynamic output
+        )
+
+
+def _determine_files_to_load(
+        bigquery_client: bigquery.client.Client,
+        target_hca_dataset: TargetHcaDataset,
+        staging_dataset: HcaScratchDatasetName,
+        file_load_table_name: str,
+        scratch_config: ScratchConfig,
+) -> RowIterator:
+    fq_dataset_id = _fully_qualified_jade_dataset_name(target_hca_dataset.dataset_name)
+    query = f"""
+    WITH J AS (
+        SELECT target_path FROM `{target_hca_dataset.project_id}.{fq_dataset_id}.datarepo_load_history` WHERE state = 'succeeded'
+    )
+    SELECT S.source_path AS sourcePath, S.target_path as targetPath
+    FROM {file_load_table_name} S LEFT JOIN J USING (target_path)
+    WHERE J.target_path IS NULL
+    """
+
+    # configure the external table definition
+    job_config = bigquery.QueryJobConfig()
+    external_config: ExternalConfig = ExternalConfig(
+        ExternalSourceFormat.NEWLINE_DELIMITED_JSON
+    )
+    external_config.source_uris = [
+        f"{_gs_path_from_bucket_prefix(scratch_config.scratch_bucket_name, scratch_config.scratch_prefix_name)}/data-transfer-requests/*"
+    ]
+    external_config.schema = [
+        SchemaField("source_path", "STRING"),
+        SchemaField("target_path", "STRING")
+    ]
+    job_config.table_definitions = {
+        "file_load_requests": external_config
+    }
+
+    # setup the destination and other configs
+    job_config.destination = f"{staging_dataset}.{file_load_table_name}"
+    job_config.use_legacy_sql = False
+    job_config.write_disposition = WriteDisposition.WRITE_TRUNCATE
+
+    # todo retry on 5xx failure (DSPDC-1729)
+    query_job: bigquery.QueryJob = bigquery_client.query(
+        query,
+        job_config,
+        location='US',
+        project=scratch_config.scratch_bq_project
+    )
+    return query_job.result()
+
+
+def _extract_files_to_load_to_control_files(
+        bigquery_client: bigquery.client.Client,
+        scratch_config: ScratchConfig,
+        scratch_dataset_name: HcaScratchDatasetName,
+        file_load_table_name: str
+) -> bigquery.ExtractJob:
+    # dump the contents of our file load requests table to a file in GCS suitable for use as a
+    # control file for jade bulk data ingest
+    out_prefix = _gs_path_from_bucket_prefix(scratch_config.scratch_bucket_name, scratch_config.scratch_prefix_name)
+    out_path = f"{out_prefix}/data-transfer-requests-deduped/*"
+
+    job_config = bigquery.job.ExtractJobConfig()
+    job_config.destination_format = bigquery.DestinationFormat.NEWLINE_DELIMITED_JSON
+    extract_job: bigquery.ExtractJob = bigquery_client.extract_table(
+        f"{scratch_dataset_name}.{file_load_table_name}",
+        destination_uris=[
+            f"{out_path}"
+        ],
+        job_config=job_config,
+        location="US",
+        project=scratch_config.scratch_bq_project
+    )
```

Comment:
i think a decent hunk of this (specifically the hardcoded values and constants) could be moved into the "job builder" resource i suggested earlier